superset/config.py

```diff
@@ -196,6 +196,13 @@ def _try_json_readsha(filepath: str, length: int) -> str | None:
 # SQLALCHEMY_DATABASE_URI = 'mysql://myapp@localhost/myapp'
 # SQLALCHEMY_DATABASE_URI = 'postgresql://root:password@localhost/myapp'
 
+# The default MySQL isolation level is REPEATABLE READ whereas the default PostgreSQL
+# isolation level is READ COMMITTED. All backends should use READ COMMITTED (or similar)
+# to help ensure consistent behavior.
+SQLALCHEMY_ENGINE_OPTIONS = {
+    "isolation_level": "SERIALIZABLE",  # SQLite does not support READ COMMITTED.
```

Comment:
@john-bodley Reading this it looks like everything will be `SERIALIZABLE` right? we're overriding it in the tests but are we overriding it as `READ COMMITTED` anywhere else for the other databases?

Reply:
@sadpandajoe followed the rabbit hole here, and I think you're right, this will bump all envs who don't set their SQLALCHEMY_ENGINE_OPTIONS to a higher isolation level.

Reply:
How about this draft? https://github.com/apache/superset/pull/30174